patches/api/0417-Add-mob-spawner-ranges-API.patch

```diff
@@ -0,0 +1,67 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: DavidTs93 <david.ts93@gmail.com>
+Date: Fri, 7 Apr 2023 19:28:25 +0300
+Subject: [PATCH] Add mob spawner ranges API
+
+
+diff --git a/src/main/java/org/bukkit/block/CreatureSpawner.java b/src/main/java/org/bukkit/block/CreatureSpawner.java
+index 5bbae759ce39d42886994e500fd9454ec328f804..dbe8e405d261b867c972b83eb923ba836d1d2e1a 100644
+--- a/src/main/java/org/bukkit/block/CreatureSpawner.java
++++ b/src/main/java/org/bukkit/block/CreatureSpawner.java
+@@ -176,31 +176,50 @@ public interface CreatureSpawner extends TileState {
+     public void setRequiredPlayerRange(int requiredPlayerRange);
+ 
+     /**
+-     * Get the radius around which the spawner will attempt to spawn mobs in.
++     * Get the horizontal radius around which the spawner will attempt to spawn mobs in.
+      * <br>
+      * This area is square, includes the block the spawner is in, and is
+      * centered on the spawner's x,z coordinates - not the spawner itself.
+      * <br>
+-     * It is 2 blocks high, centered on the spawner's y-coordinate (its bottom);
++     * It is {@link #getSpawnRangeVertical()} + 1 blocks high, centered on the spawner's y coordinate (its bottom);
+      * thus allowing mobs to spawn as high as its top surface and as low
+-     * as 1 block below its bottom surface.
++     * as {@link #getSpawnRangeVertical()} blocks below its bottom surface.
+      * <br>
+      * Default value is 4.
+      *
+-     * @return the spawn range
++     * @return the horizontal spawn range
+      */
+     public int getSpawnRange();
+ 
+     /**
+-     * Set the new spawn range.
++     * Set the new horizontal spawn range.
+      * <br>
+      *
+-     * @param spawnRange the new spawn range
++     * @param spawnRange the new horizontal spawn range
+      * @see #getSpawnRange()
+      */
+     public void setSpawnRange(int spawnRange);
+ 
+     // Paper start
++    /**
++     * Get the vertical radius around which the spawner will attempt to spawn mobs in.
++     * <br>
++     * Default value is 1.
++     *
++     * @return the vertical spawn range
++     * @see #getSpawnRange()
++     */
++    public int getSpawnRangeVertical();
```

Comment:
I think "VerticalSpawnRange" sounds better than SpawnRangeVertical. Its more correct English. Same for all the mentions of those 3 words. Best to stick to the more grammatically correct version.